### ISSUE TITLE
Short form for console output

### ISSUE THREAD

**Issue by lweberk** [philsquared/Catch#378]
Hi,

it would be great to have a optionally shorted output form like:

Scenario: myscenario
         Given: certain conditions 1
         Given: certain conditions 1
             When: certain premise X [PASSED]
             When: certain premise Y [PASSED]
         Given: certain conditions 2
             When: certain premise Z [PASSED]
...
Analogously for the TDD approach.

Or without grouping them and only getting rid of the separator lines and expansions.

Or maybe only a "extended" summary? Leaving -s intact... ?

The reason is, that if you have automated testing running on a server you are foremost interested in pinpointing the one that failed, before seing why it actually failed.

Maybe I can get to this one myself, if we can agree over a specific way and output (maybe not even as mentioned above, but along the lines of it).

Thanks for all your terrific job at this project, it has turned out to be extremely useful to us.
Best regards,
~lweberk

**Comment by philsquared** [comment]
Hi @lweberk,

Thanks for the kind words. I'm curious why you feel this would be needed, given your reasoning about pinpointing which test failed quickly from build server output. Or rather I'm curious why the current formatting doesn't do it for you?

**Comment by lweberk** [comment]
Hi,

To put it in as few words as possible; too much clutter. The longer version (sorry for the formatting, can't find a way around markdown going haywire):

```
i.E. looking at (output of "-s"):
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
run_catch is a Catch v1.0 b10 host application.
Run with -? for options

-------------------------------------------------------------------------------
Connection can be established
-------------------------------------------------------------------------------
build.tests/TestConnection.cpp:12
...............................................................................

build.tests/TestConnection.cpp:21: 
PASSED:
  REQUIRE( conn.is_connected() )
with expansion:
  true

-------------------------------------------------------------------------------
Scenario: data can be queried, inserted and updated
     Given: a new object at pre-insert
      When: accessing a member without previously being set
-------------------------------------------------------------------------------
build.tests/TestDatalayer.cpp:60
...............................................................................

build.tests/TestDatalayer.cpp:76: 
PASSED:
  REQUIRE_THROWS_AS( name << client.name )

-------------------------------------------------------------------------------
Scenario: data can be queried, inserted and updated
     Given: a new object at pre-insert
      When: inserted without previously setting non-serial pkeys
-------------------------------------------------------------------------------
build.tests/TestDatalayer.cpp:60
...............................................................................

build.tests/TestDatalayer.cpp:80: 
PASSED:
  REQUIRE_THROWS_AS( conn << client )
```
I'm running these tests for the Nth time, and would only require expansion details if it failed, otherwise It just hinders me from seeing those who passed even if they very explicitly say so. Its just, that the eye needs to pick out the relevant information from a very verbose structure.

Lets hipothetically consider a more condensed output telling me the same while sparing me details (taken from the second test output above):
Scenario: data can be queried, inserted and updated
     Given: a new object at pre-insert
      When: accessing a member without previously being set  <<< imagine in green (because it is the actual passed test while "scenario" and "given" are only the premise). Also imagine without separator hlines. It would look like:

```
Scenario: data can be queried, inserted and updated
     Given: a new object at pre-insert
      When: accessing a member without previously being set
Scenario: data can be queried, inserted and updated
     Given: a new object at pre-insert
      When: inserted without previously setting non-serial pkeys
```
Imagine "When" statement preceeded lines in green. If a test fails (only then), imagine hlines and expansion as is right now.

This way you can ignore what is to be ignored and focus on the actual error. Currently there is little difference to tell them apart other than the green "PASSED" embedded within 11 lines of visually uncoded information. Which stays almost the same (11 lines) in a failed case.

On the other hand, the alternative output does not show anything unless it fails. In our case we want to use the output to pass on the scenarios to the tech-savy-but-not-devs, so they are up to date on the requirements they want seen are met and logically sound. Bascially BDD system testing with output as communication interface to pseudo-customers.

Please take this as an analisis of introspecting how my brain and eyes try to go through the output, and not criticism. One can get used to anything and deal with any type of information output (so the current output actually does for me), I just have the feeling that there is still room for improvement to achieve an easier "seeing without having to look" and a way to use the same output to communicate specifications of what you are building.

**Comment by jbrwilkinson** [comment]
Seems like you'd like a 'quiet' mode that only outputs anything at all when there's a problem?

**Comment by lweberk** [comment]
Nope, I mean a short form. Not staying quiet, but displaying if it passed or not in only 1 line. In case of test scenarios or BDD, without repeating the scenario/given part;

Scenario: scenario1...
     Given: under condition 1...
     When: some situation 1
     When: some situation 2
     When: some situation 3
...
